hydra-node/test/Hydra/Chain/Direct/TxSpec.hs

```diff
@@ -141,6 +148,58 @@ spec =
                         & counterexample (renderTx tx)
                         & counterexample (show e)
 
+    describe "commitTx" $ do
+      prop "Validate blueprint and commit transactions" $ \utxoWithWitnesses -> do
+        forAll arbitrary $ \chainContext -> do
+          let ChainContext{networkId, ownVerificationKey, ownParty, scriptRegistry} =
+                chainContext{ownVerificationKey = getVerificationKey commitSigningKey, networkId = testNetworkId}
+          let utxoToCommit = utxoFromTxOutWithWitness utxoWithWitnesses
+          let blueprintTx' = mkBlueprintTx utxoWithWitnesses
+          let tx' =
+                commitTx
+                  networkId
+                  scriptRegistry
+                  (mkHeadId Fixture.testPolicyId)
+                  ownParty
+                  utxoToCommit
+                  blueprintTx'
+                  (healthyInitialTxIn, toUTxOContext healthyInitialTxOut, verificationKeyHash ownVerificationKey)
+          let signedCommitTx = signTx commitSigningKey tx'
+          let blueprintTx = toLedgerTx blueprintTx'
+          let blueprintBody = blueprintTx ^. bodyTxL
+          let tx = toLedgerTx tx'
+          let commitTxBody = tx ^. bodyTxL
+          let containsScript u = any (isJust . witness . snd) (UTxO.pairs u)
+
+          let transactionEvaluates (transaction, lookupUTxO) =
```

Comment:
Should: use existing `propTransactionEvaluates` (and maybe move this out of `Hydra.Chain.Direct.Contract.Mutation`)